sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -566,7 +566,15 @@ private static Field getFieldBasedOnRexFieldAccess(
 
   private Pair<RexNode, RexNode> extractJoinPairOfRexNodes(RexCall rexCall) {
     if (!rexCall.getOperator().getName().equals("=")) {
-      throw new UnsupportedOperationException("Non equi-join is not supported!");
+      throw new UnsupportedOperationException("Non equi-join is not supported");
+    }
+
+    if ((!(rexCall.getOperands().get(0) instanceof RexInputRef)
```

Comment:
Can you wrap these conditions into methods like `isFieldOrInputRef()` with a comment? Otherwise it's hard to read